io/plugins/eu.esdihumboldt.hale.io.gml.test/src/eu/esdihumboldt/hale/io/gml/writer/internal/StreamGmlWriter2Test.groovy

```diff
@@ -381,4 +381,72 @@ class StreamGmlWriter2Test {
 
 		result
 	}
+
+	@Test
+	void testNumbers() throws Exception {
+		double doubleNumber = 123456789.123456789
+		double expected = java.lang.Double.parseDouble(doubleNumber.toString())
+
+		testNumber("doubleSimpleNumber", doubleNumber, expected)
+
+		testNumber("integerNumber", 12345, null)
+		testNumber("doubleNumber", 12345.123456789, null)
+		testNumber("floatNumber", 12345.6789F, null)
+		testNumber("longNumber", 1234567890123456789L, null)
+		testNumber("shortNumber", 12345 as short, null)
+		testNumber("decimalNumber", new BigDecimal("1234567890.123456789"), null)
```

Comment:
I think it would be better to really explicitly use an expected string representation instead of relying on the toString representation. With that we could be sure that the representation is as expected as string and not rely on toString to yield the values as expected. After all the issue was that `toString` was used before and it did not yield the desired values at least for the `decimal` type.

Reply:
I have removed the `toString` and noticed that [double and float are treated in the same way](https://github.com/halestudio/hale/blob/43771301efa9df1f02e8822f86ae313a63e16308/io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/writer/internal/StreamGmlWriter.java#L1624) I have add more tests for this 2 types.

Reply:
I'm afraid I think you misunderstood me. It is important to check the actual text used in the XML representation, so my suggestion is to use concrete strings as expected values, to be sure the notation is as expected.